CHANGELOG.md

```diff
@@ -3,6 +3,37 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [6.0.0] - TBD
+### Added
+- Per-schema option to send S3 data notifications to an SQS queue.  See `enable_data_event_queue` in the [apiary_managed_schemas](VARIABLES.md#apiary_managed_schemas) section of [VARIABLES.md](VARIABLES.md)
+### Changed
+- Changed AWS resources created on a per-schema basis to use Terraform `for_each` instead of `count`.  This includes S3 and SNS resources.
+  - This was done to fix the issue of removing a schema in a later deployment.  If the schema removed is not at the end of the `apiary_managed_schemas` list, then when using `count`, Terraform will see different indexes in the state file for the other resources, and will want to delete and recreate them. Using `for_each` references them by `schema_name` in the state file and fixes this issue.
+### Notes
+- *THIS IS A BREAKING CHANGE.* When deploying `6.0.0` on an existing Apiary deployment, the following procedure must be followed:
+  - See the `migrate.py` script in the `scripts` folder.
+  - This script is used to migrate an Apiary Terraform state file from using `count` for resource indexing to using
+      `for_each`, which is how apiary-data-lake v6.0.0+ handles indexed resources.  Without this script, doing an `apply` 
+      will want to destroy all your S3 resources and then recreate them because they are stored in the `.tfstate` file
+      differently.  
+  - The migration script needs some external packages installed (see `migrate_requirements.txt`) and then should run in either Python 2.7+ or Python 3.6+.
+  - This procedure assumes you have a Terraform app called `apiary-terraform-app` that is the application using this module.
+  - Upgrade `apiary-terraform-app` to `apiary-data-lake` v5.3.2.  This will necessitate using Terraform 0.12+ and
+    resolving any TF 0.12 incompatibilities in your application code.  TF 0.12.21+ is recommended (will be required later).
+  - Plan and apply your Terraform app to make sure it is working and up-to-date.
+  - Install Python 3 if you don't yet have a Python installation.
+  - Install requirements for this script with `pip install -r migrate_requirements.txt`.
+  - Run this script pointing to your terraform state file.  Script can read the state file from either file system or S3. Run it first with dryrun, then live.  Example:
+    - `python migrate.py --dryrun --statefile s3://terraform-state-672016901104-us-east-1/apiary/datalake/terraform.tfstate`
```

Comment:
What is the number in the bucket?

Reply:
I just pasted in the `egdp-dev` bucket/object as the example.  Would you rather see a more generic path in the example?

Reply:
I changed the example to a more generic one.